src/wrapper/quaserver.cpp

```diff
@@ -1478,7 +1478,7 @@ UA_Logger QUaServer::getLogger()
 			vsprintf(srv->m_logBuffer.data(), msg, args);
 			// NOTE : do not convert QBytearray to QString to avoid overhead
 			emit srv->logMessage({
-				srv->m_logBuffer,
+				QString::fromUtf8( srv->m_logBuffer.data() ),
```

Comment:
Thanks for reporting the log issue and the fix proposal, although did you consider the comment above?
> // NOTE : do not convert QBytearray to QString to avoid overhead

It was a long time ago, but I remember that logging was hurting performance due to convertion to `QString`. Do we have any other alternative? Maybe refactor `logMesaage` [to use `QStringView`?](https://doc.qt.io/qt-5/qstringview.html#details)

Reply:
You can not convert to QString, but leave QByteArray, but then you need to refactor the QUaLog structure. And there someone left a comment: "consutructor accepting QString instead of QByteArray (to support generating messages with QObject::tr)". Probably you need to add the constructor QUaLog(const QByteArray& ... I'll try it in the evening.

Reply:
By the way, in many places there is an implicit conversion to QString, you just don't see it. I'll make these transformations explicit later so you can see them.

Reply:
I bielieve you 😉 , if you make a refactor that is as efficient or more than the state of the art, I will be very happy to merge it ⚡

Reply:
Looks good, so will merge. Disclaimer; I am not testing it.

Reply:
I tested on Linux x86-64 (Qt 5.15.9), Windows10 x86-64 (Qt 5.15.9, 6.5.1).

Reply:
Excellent, thank you for the fix